### PR TITLE
Conversions: Makes conversion queries more flexible

### PR DESCRIPTION
**Pull request by adityatandon007**
**Conversions IA**: This PR makes the queries flexible as mentioned in comment of #4118 

## Description of new Instant Answer, or changes
Improves triggering

## Related Issues and Discussions
Continuing #4118 

## People to notify
@moollaza @mintsoft

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/conversions

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by adityatandon007** [comment]
@moollaza @mintsoft I have deleted the test for seconds and minutes because we have a countdown timer IA which triggers for `seconds` or `minutes` and so it was conflicting with those.

File: t/Conversions.t [hunk 1]
```diff
@@ -3375,9 +3412,7 @@ ddg_goodie_test(
     'use a ton of stones'             => undef,
     'shoot onself in the foot'        => undef,
     'foot in both camps'              => undef,
-    'Seconds'                         => undef,
```
Comment (mintsoft) [inline t/Conversions.t]: Hmm, these still shouldn't trigger, have you removed them because they are now?

File: lib/DDG/Goodie/Conversions.pm [hunk 2]
```diff
@@ -42,7 +42,7 @@ my $question_prefix = qr/(?<prefix>convert|what (?:is|are|does)|how (?:much|many
 # guards and matches regex
 my $factor_re = join('|', ('a', 'an', number_style_regex()));
 
-my $guard = qr/^(?<question>$question_prefix)\s?(?<left_num>$factor_re*)\s?(?<left_unit>$keys)\s((equals|=|is)\s((how many )|(\? ))?)?(?<connecting_word>in|to|into|(?:in to)|from)?\s?(?<right_num>$factor_re*)\s?(?:of\s)?(?<right_unit>$keys)\s?(?:conversion)?[\?]?$/i;
+my $guard = qr/^(?<question>$question_prefix)\s?(?<left_num>$factor_re*)\s?(?<left_unit>$keys)\s?((equals|=|is)\s?((how many)|(\?)\s?)?)?(?<connecting_word>in|to|into|(?:in to)|from)?\s?(?<right_num>$factor_re*)\s?(?:of\s)?(?<right_unit>$keys)\s?(?:conversion)?[\?]?$/i;
```
Comment (mintsoft) [inline lib/DDG/Goodie/Conversions.pm]: I think we probably need to keep the whitespace around `equals` and `is`, however it should probably be optional for the =.
Maybe something similiar to (untested!): `(?<left_unit>$keys)((\s(equals|is)\s)|(\s?=\s?))((how many`

**Comment by mintsoft** [comment]
@adityatandon007 

> I have deleted the test for seconds and minutes because we have a countdown timer IA which triggers for seconds or minutes and so it was conflicting with those.

The tests for conversions will have no impact on the tests for countdown; that said it is important that we don't trigger on those queries, unless you've changed the behaviour I think the tests should remain

**Comment by adityatandon007** [comment]
@mintsoft Thanks I'll look once again on that regex and then let you know if I get stuck

**Comment by adityatandon007** [comment]
@mintsoft and @moollaza I have added that tests back and it is not triggering for those test as mentioned in the test file. One more thing the query triggers for everything except `190lb=?kg`. I think it shouldn't because for that we have to make space between `lb` and `=` optional and because of which queries like `seconds` and `minutes` gets triggered and therefore it will change the whole regex and the rest of the 314 test cases. I don't think it w'd be good to make changes to all the others queries behavior just to add a single flexible query to trigger. Queries like `190lb =?kg` and others are triggering fine. Just look at the regex and you will understand what I am trying to imply here :slightly_smiling_face:

**Comment by mintsoft** [comment]
@adityatandon007 Yeah that's fine, we can add `190lb=?kg` separately if we can find a nice way of adding it

**Comment by mintsoft** [comment]
@moollaza this LGTM

**Comment by moollaza** [comment]
@adityatandon007 @mintsoft Thanks! This LGTM :+1:

**Comment by bsstoner** [comment]
This was released to production 🚀

**Pull request merged**